bpf: use syscall.BytePtrFromString instead of deprecated syscall.StringBytePtr

tklauser opened the pull request:
`syscall.StringBytePtr` is deprecated and its godoc comment suggests using
`syscall.BytePtrFromString` instead. This will return a proper error
instead of panic'ing.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10117)
<!-- Reviewable:end -->

rolinh commented on pkg/bpf/bpf_linux.go:
```diff
@@ -272,7 +272,11 @@ type bpfAttrObjOp struct {
-	pathStr := syscall.StringBytePtr(pathname)
+	pathStr, err := syscall.BytePtrFromString(pathname)
+	if err != nil {
+		return fmt.Errorf("Unable to convert pathname %q to byte pointer: %v", pathname, err)
```
I think that it would be useful to wrap the underlying error (using `%w` instead of `%v`) as it can be of type errno and it can be useful to be able to check that.

tklauser replied:
Good point, will change.

rolinh replied:
I started removing explicit "errno functions" in https://github.com/cilium/cilium/pull/10058 but I think that this strategy of wrapping errno errors should be replicated to ease debugging/checking specific errors.

rolinh commented on pkg/bpf/bpf_linux.go:
```diff
@@ -301,7 +305,11 @@ func ObjPin(fd int, pathname string) error {
-	pathStr := syscall.StringBytePtr(pathname)
+	pathStr, err := syscall.BytePtrFromString(pathname)
+	if err != nil {
+		return -1, fmt.Errorf("Unable to convert pathname %q to byte pointer: %w", pathname, err)
```
I think that `-1` should be `0` here considering it's the default value and how it's used on non-nil return path in this function.

tklauser replied:
Thanks, done.

tklauser committed:
bpf: use syscall.BytePtrFromString instead of deprecated syscall.StringBytePtr

syscall.StringBytePtr is deprecated and its godoc comment suggests using
syscall.BytePtrFromString instead. This will return a proper error
instead of panic'ing.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
